### PR TITLE
buildah image: install cpp

### PR DESCRIPTION
**Pull request by vrothberg**
Ship `cpp` with the Buildah container images to make sure that
preprocessing .in files works as expected and documented.

Also improve the error message when cpp isn't installed.

Fixes: #3822

### CODE REVIEW

**Review by flouthoc — APPROVED** [review]
LGTM. @vrothberg would this update `:latest` as well.

**Comment by vrothberg** [comment]
> LGTM. @vrothberg would this update `:latest` as well.

Yes, all but the "manual" one are built by Quay automatically. @TomSweeneyRedHat is their loving shepherd.

**Comment by TomSweeneyRedHat** [comment]
@vrothberg and probably @cevich, here's the question I don't want to ask.  Do we need to rebuild the current versions in quay.io to make this work for everyone?  Or do we add a note to the README.md telling folks about the problem, which versions of the images were affected, and how (if possible) to fix it for a particular image?

**Comment by TomSweeneyRedHat** [comment]
The changes LGTM, btb, unless a README.md adjustment is needed.

**Comment by cevich** [comment]
> Do we need to rebuild the current versions in quay.io to make this work for everyone? Or do we add a note to the README.md telling folks about the problem, which versions of the images were affected, and how (if possible) to fix it for a particular image?

Oof, maybe both, checking...

**Comment by cevich** [comment]
...what should happen once this is merged:  Within a day the multi-arch images and manifest list will be built including these changes.  For 'testing' and 'upstream', it will push to the 'latest' tagged images only.  For 'stable', both 'latest' and the version-tagged images will be updated.  Previously pushed (older) version-tagged images won't be affected at all.

So this could cause unexpected changes in the most recent version-tagged 'stable' image.  That's probably where a README update might be nice, making note of the version-tag where the change appears.

Hope that helps.

Ref: https://github.com/containers/buildah/blob/main/.github/workflows/multi-arch-build.yaml

**Comment by vrothberg** [comment]
Which README are we talking about?  I don't see one where we record bug fixes.  I would keep it simple, merge and then update the images.

**Comment by TomSweeneyRedHat** [comment]
If we add to the readme.md, it would be this one: https://github.com/containers/buildah/tree/main/contrib/buildahimage

**Pull request merged**
